### PR TITLE
base code for clm-ocr adapter

### PR DESCRIPTION
**Pull request by kundan-sf**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/contract-ocr/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Check out https://hub.docker.com/_/node to select a new base image
```
Comment (akshatdubeysf) [inline packages/contract-ocr/Dockerfile:1]: Not required for a package

File: packages/contract-ocr/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# ocr
+
```
Comment (akshatdubeysf) [inline packages/contract-ocr/README.md:2]: Put actual readme and description

File: packages/contract-ocr/package.json [hunk 3]
```diff
@@ -0,0 +1,99 @@
+{
+    "name": "ocr-ocr",
```
Comment (akshatdubeysf) [inline packages/contract-ocr/package.json:2]: proper name with sourceloop prefix

File: packages/contract-ocr/package.json [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        "@loopback/eslint-config": "^12.0.4",
+        "@loopback/testlab": "^4.1.2",
+        "@types/node": "^12.20.47",
+        "dotenv": "^16.0.0",
+        "dotenv-extended": "^2.9.0",
+        "eslint": "^8.10.0",
+        "nyc": "^15.1.0",
+        "source-map-support": "^0.5.21",
+        "typescript": "~4.6.3"
+    }
```
Comment (akshatdubeysf) [inline packages/contract-ocr/package.json:98]: add private: true

File: packages/contract-ocr/public/index.html [hunk 5]
```diff
@@ -0,0 +1,103 @@
+<!DOCTYPE html>
```
Comment (akshatdubeysf) [inline packages/contract-ocr/public/index.html:1]: not required in a package

File: packages/contract-ocr/src/component.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+} from './keys';
+import {
+    RequestProvider
+} from './providers';
+
+export class RequestComponent implements Component {
+    constructor() {}
+
+    providers?: ProviderMap = {
+        [RequestBindings.FetchProvider.key]: RequestProvider
```
Comment (akshatdubeysf) [inline packages/contract-ocr/src/component.ts:17]: should bind both the providers that are provided by the component?

File: packages/contract-ocr/src/keys.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import {
+    BindingKey
+} from '@loopback/core';
+import {
+    HttpClientInitOpts,
+    IRequest
+} from './types';
+
+export namespace RequestBindings {
```
Comment (akshatdubeysf) [inline packages/contract-ocr/src/keys.ts:9]: Change the to something more specific

File: packages/contract-ocr/src/keys.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    export const RequestProvider = BindingKey.create <IRequest > ('sf.request');
+    export const FetchProvider = BindingKey.create<IRequest>('sf.notification.fetch');
+    export const Config = BindingKey.create < HttpClientInitOpts | null> ('sf.request.config');
```
Comment (akshatdubeysf) [inline packages/contract-ocr/src/keys.ts:12]: formatting

File: packages/contract-ocr/src/opentelemetry-registry.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import {JaegerExporter} from '@opentelemetry/exporter-jaeger';
```
Comment (akshatdubeysf) [inline packages/contract-ocr/src/opentelemetry-registry.ts:1]: this file is not required in a package

File: packages/contract-ocr/src/providers/fetchClient/fetch-client.provider.ts [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+        const jsonHeaders = useJson ? {
+                [Header.Accept]: "application/json",
+                [Header.ContentType]: "application/json",
+            } :
+            undefined;
+
+        this.baseUrl = baseUrl;
+
+        this.baseHeaders = {
+            ...jsonHeaders,
```
Comment (akshatdubeysf) [inline packages/contract-ocr/src/providers/fetchClient/fetch-client.provider.ts:76]: would throw error if jsonHeaders is undefined

File: packages/contract-ocr/src/providers/fetchClient/fetch-client.provider.ts [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+        this.transformResponse = useJson ?
+            jsonResponseTransformer :
+            identityResponseTransformer;
+
+        this.useJson = useJson;
+    }
+    value(): ValueOrPromise < FetchHttpRequest > {
+        return {
+            send: async (url: string, req: RequestInit) => {
+                if (req.method?.toUpperCase() === 'POST') {
```
Comment (akshatdubeysf) [inline packages/contract-ocr/src/providers/fetchClient/fetch-client.provider.ts:94]: only post requests allowed?

File: packages/contract-ocr/src/providers/fetchClient/fetch-client.provider.ts [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+                    return this.post(url, req)
+                }
+            }
+        };
+    }
+
+    async get < T > (url: string, req: RequestInit = {}): Promise < T > {
```
Comment (akshatdubeysf) [inline packages/contract-ocr/src/providers/fetchClient/fetch-client.provider.ts:101]: what is the purpose of this then?

File: packages/contract-ocr/src/providers/index.ts [hunk 13]
```diff
@@ -0,0 +1,2 @@
+export * from './fetchClient';
```
Comment (akshatdubeysf) [inline packages/contract-ocr/src/providers/index.ts:1]: folder name patter is incorrect

File: packages/contract-ocr/src/providers/types.ts [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+export type RequestInterceptor = (
+  url: string,
+  request: RequestInit,
+) => void | Promise<void>;
+
+export type ResponseTransformer = (res: Response) => any;
+
+export type HttpClientInitOpts = {
```
Comment (akshatdubeysf) [inline packages/contract-ocr/src/providers/types.ts:15]: these values appear in two files?

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

File: packages/contract-ocr/Dockerfile (hunk 1 above)
Comment (kundan-sf) [reply packages/contract-ocr/Dockerfile]: Okay done in my latest PR

File: packages/loopback4-fetch-client/src/providers/fetchClient/fetch-client.provider.ts [hunk 15]
```diff
@@ -0,0 +1,206 @@
+import {inject, Provider, ValueOrPromise} from '@loopback/core';
```
Comment (akshatdubeysf) [inline packages/loopback4-fetch-client/src/providers/fetchClient/fetch-client.provider.ts:1]: why the same provider twice?

**Pull request closed without merging**